client/src/lib.rs

```diff
@@ -361,6 +371,13 @@ impl ClientOptions {
 
             if let Some(domain) = &tls_cfg.domain {
                 tls = tls.domain_name(domain);
+
+                // This song and dance ultimately is just to make sure the `:authority` header ends
+                // up correct on requests while we use TLS. Setting the header directly in our
+                // interceptor doesn't work since seemingly it is overridden at some point by
+                // something lower level.
+                let uri: Uri = format!("https://{}", domain).parse()?;
+                channel = channel.origin(uri);
```

Comment:
Do we need to override this when `domain` is set? Can't we leave the origin untouched?

Reply:
That's literally the whole fix

Reply:
Oh is it just for the case where `domain` is different than the URL?

Reply:
Exactly

Reply:
I see, we call it `serverNameOverride` in TS so I was confused by the name.